tests.php

```diff
@@ -42,6 +42,7 @@
 	// Misc
 	$a->withHeader('Access-Control-Allow-Origin', '*');
 	include('abc.xyz');
+	require_once EXTENSION_PATH . '/path/to' . $name . '.php';
```

Comment:
the test file is actually negative testing - meaning that they are supposed to trigger the rule top fail instead of passing. so using `.php` doesn't trigger it, I'll change this to another file name then merge.

Reply:
The unit test shouldn't be changed.

Reply:
Test this by running `phpcs ./tests.php --standard=Security` on the adjusted unit test file _without_ this fix.

Without the fix, you will receive the below message:
```
45 | ERROR   | No file extension has been found in a include/require function. This implies that some PHP code is not scanned by PHPCS.
```

The fix is about not throwing this incorrect message for valid file extensions which are being checked by the scan.

Reply:
Sorry I already merged, but no problem I'll change back if needed.

Let me figure this out.

Reply:
In other words, it's about preventing a false positive.

Reply:
Ok I just ran it like you said and I think I got it.

The good news is that you'll see the same errors as you mention here even if the file extension is jkl. And having the extension to jkl does align with what I was saying about having the tests.php file act negatively on the change.

Let me know if you still think the change to jkl is detrimental to your fix.

Reply:
@jmarcil Yes, I still think so.

With `jkl`, you would expect an error on that line. The fact that it's being triggered on the wrong string `path/to` vs `.jkl` doesn't matter. You get an error, even though the actual error you get is different before and after the fix.

With the `php` extension, there shouldn't be an error on that line, while without my fix, there is one.

Reply:
Either way, moving to more sniff-specific unit tests which would test for both positives, false positives, negatives and false negatives would help with all this.

Reply:
So it appears from your end that I'm destroying the false positive check by replacing it with a true negative check?

Here's what I see:
with .php in the test, before the sniff fix:
```
 45 | ERROR   | No file extension has been found in a include/require function. This implies that some PHP code is not scanned by PHPCS.
```
with .jkl in the test, before the sniff fix:
```
 45 | ERROR   | No file extension has been found in a include/require function. This implies that some PHP code is not scanned by PHPCS.
```

So the false positive error is the same regardless of the file type.

Now with the fix in place, testing for `.php` doesn't give me any results, while `.jkl` gives me:
```
 45 | ERROR   | The file extension '.jkl' that is not specified by --extensions has been used in a include/require function. Please add it to the
```

From what I can tell everything looks fine for both ways of reasoning.

But yes, we can tackle the test the way you wanted it to behave into sniff-specific unit tests, but for now I would like to respect the ways of doing of this tool in `tests.php` even if it isn't perfect.

Thanks for your understanding, now if you allow me I really want to put my time into #50.

Reply:
Fair enough and yes, please ;-)